### PR TITLE
Map block: show preview not available in customizer to prevent js error

### LINKED ISSUE #19976: Map block: block widget doesn't show in customizer preview
The Mapblock can be inserted and previewed in the block widget panel, but does not display in the customizer preview.

#### Steps to reproduce the issue

1. In local dev run latest Jetpack and gutenberg master/trunk and latest built of 5.8 core
2. Go to customizer widgets on a non-block theme, eg. tt1 standard and add a map block block. Notice that there is no preview of the block on the right

**Screenshots**
Screen Shot 2021-05-29 at 4 24 36 PM

### PR DESCRIPTION
**Pull request by glendaviesnz**
Fixes #19976

#### Changes proposed in this Pull Request:

* If the Map block is added as a widget to the customizer it fails to load and causes an exception in the console. This is because the `wp-element` dependency is not loaded by default in the customizer, and it doesn't see possible to get these dependencies loaded dynamically in the customizer iframe. This PR instead indicates to the user that a preview is not available for that widget.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Check out this PR and run locally with latest Gutenberg and latest WP dev build
* In customizer add a map widget and check there are no errors in console and that the 'no preview' message displays

Before:
before

After:
after

### CODE REVIEW

File: projects/plugins/jetpack/class.jetpack-gutenberg.php
Comment (andrewserong) [inline projects/plugins/jetpack/class.jetpack-gutenberg.php:629]: Tiniest of tiny nits: should we add another full-stop at the end of the sentence?

**Review by andrewserong — COMMENTED** [review]
I think this is a good pragmatic approach for this case @glendaviesnz. I noticed that the Map block's fallback markup is still being rendered, though. So if I have a couple of map markers added, the links of text for places on the map are still being rendered in the preview:

image

Did we want to hide these, too?

File: projects/plugins/jetpack/class.jetpack-gutenberg.php
Comment (apeatling) [reply projects/plugins/jetpack/class.jetpack-gutenberg.php:629]: I agree, should have one at the end.

Comment (apeatling) [reply projects/plugins/jetpack/class.jetpack-gutenberg.php:629]: I think it should also say "No map preview available". To be clearer in situations where there are multiple blocks.

Comment (glendaviesnz) [reply projects/plugins/jetpack/class.jetpack-gutenberg.php]: Done.

**Comment by glendaviesnz** [comment]
> Did we want to hide these, too?

Have added a some extra script to do this.

@andrewserong, @apeatling this should be ready for another review

**Review by andrewserong — DISMISSED** [review]
Looks good to me @glendaviesnz, thanks for adding in the clever line to hide fallback map markup, too 👍

image

**Review by jeherve — DISMISSED** [review]

**Comment by sdixon194** [comment]
@glendaviesnz looks like there's a conflict that needs to be resolved before we can merge! 

Also forced the wpcom teamcity test to pass, so the wpcom diff will need to be handled manually. 👍

**Review by ramonjd — APPROVED** [review]
I think this is a good interim solution.

I'm just jotting down what I'm seeing for future reference:

When I add a map to my footer widgets area, the map loads and _kinda_ works until I interact with it, upon which it disappears.

mappy

For me it's _slightly_ confusing as the copy says no preview is available.

We could prevent this by not loading the map's dependencies during autosave.

E.g.,

```php

			if ( is_customize_preview() && ! empty( $_GET['customize_autosaved'] ) ) {
				if ( 'map' === $type ) {
					echo '<div>' . esc_html_e( 'No map preview available. Publish and refresh to see this widget.', 'jetpack' ) . '</div>';
					...
				} else {
					echo '<script id="jetpack-block-' . esc_attr( $type ) . '" src="' . esc_attr( $view_script ) . '?ver=' . esc_attr( $script_version ) . '"></script>';
				}

				// Enqueue dependencies, but not for map as it will cause a non-interactive map to show.
				if ( 'map' !== $type ) {
					wp_enqueue_script( 'jetpack-block-' . $type, $view_script, $script_dependencies, $script_version, false );
				}
			} else {
				wp_enqueue_script( 'jetpack-block-' . $type, $view_script, $script_dependencies, $script_version, false );
			}

```

Also, when I interact with the map after publish + preview, it disappears again.

disappears_wut

This might be a crazy idea, but, if run into this sort of stuff again, maybe we could experiment with publishing an invisible widget in the editor - one that is always in the customizer widget editor but never appears on the frontend, and also pulls in the dependencies we need for these widgets. 

This idea might work since, during my testing, I noticed that I could in fact load the map when other widgets with shared dependencies were already published and loaded in the preview. 

This is a long way of saying LGTM 👍

**Comment by glendaviesnz** [comment]
@jeherve or @sdixon194, sorry the conflict resolution in the rebase forced the need for another review - feel free to go ahead and merge this.

Manual diff added at D64229-code

**Review by sdixon194 — APPROVED** [review]

**Comment by sdixon194** [comment]
Cherry picked to release branch in e3dcd2b43de06e226188e2f413129a5e416efbc0

**Pull request merged**
